File: BaseBackend/src/main/java/com/common/dto/response/AuthResponse.java
package com.common.dto.response;

import com.common.model.enumerate.UserRole;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.Set;

/**
 * DTO phản hồi kết quả xác thực Auth (Access Token, Refresh Token, User context).
 *
 * @created_at 2026-07-31
 * @author txhoan
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AuthResponse {

    String accessToken;

    String refreshToken;

    @lombok.Builder.Default
    String tokenType = "Bearer";

    long expiresIn;

    Long userId;

    String username;

    String email;

    Long tenantId;

    String tenantCode;

    Set<UserRole> roles;
}

File: be/.Net/Swe.BL/DTO/Response/AuthResponse.cs
using System.Collections.Generic;

namespace Swe.BL.DTO.Response;

/// <summary>
/// DTO phản hồi kết quả xác thực Auth.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
public class AuthResponse
{
    public string AccessToken { get; set; } = string.Empty;
    public string RefreshToken { get; set; } = string.Empty;
    public string TokenType { get; set; } = "Bearer";
    public long ExpiresIn { get; set; }
    public long UserId { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public long? TenantId { get; set; }
    public string? TenantCode { get; set; }
    public List<string> Roles { get; set; } = new();
}
